vview/src/application.rs

```diff
@@ -0,0 +1,344 @@
+        let RenderState::Suspended(cached_window) = &mut self.render_state else {
+            return;
+        };
+
+        let window = cached_window
+            .take()
+            .unwrap_or_else(|| create_window(&event_loop));
+
+        let size: winit::dpi::PhysicalSize<u32> = window.inner_size();
+        let render_ctx = &mut self.render_ctx;
+        let surface_future = render_ctx.create_surface(window.clone(), size.width, size.height, wgpu::PresentMode::AutoVsync);
+        
+        // We need to block here, in case a Suspended event appeared
+        let surface: RenderSurface = pollster::block_on(surface_future).expect("Error creating surface");
+
+        self.render_state = {
+            self.renderers.resize_with(render_ctx.devices.len(), || None);
+            self.renderers[surface.dev_id].get_or_insert_with(||create_vello_renderer(&render_ctx, &surface));
+            RenderState::Active(ActiveRenderState { window, surface })
+        };
+        event_loop.set_control_flow(ControlFlow::Poll);
+    }
+
+    fn suspended(&mut self, event_loop: &ActiveEventLoop) {
+        if let RenderState::Active(state) = &self.render_state {
+            self.render_state = RenderState::Suspended(Some(state.window.clone()));
+        }
+        event_loop.set_control_flow(ControlFlow::Wait);
+    }
+
+    fn window_event(&mut self, event_loop: &ActiveEventLoop, window_id: WindowId, event: WindowEvent) {
+        // Ignore the event (return from the function) if
+        //   - we have no render_state
+        //   - OR the window id of the event doesn't match the window id of our render_state
+        //
+        // Else extract a mutable reference to the render state from its containing option for use below
+        let render_state = match &mut self.render_state {
+            RenderState::Active(state) if state.window.id() == window_id => state,
+            _ => return,
+        };
+        let render_ctx = &self.render_ctx;
+
+        match event {
+            WindowEvent::CloseRequested => event_loop.exit(),
+            WindowEvent::ModifiersChanged(m) => self.modifiers = m.state(),
+            WindowEvent::KeyboardInput { event, .. } => {
+                if event.state == ElementState::Pressed {
+                    if self.modifiers.shift_key() {
+                        match event.logical_key {
+                            Key::Named(NamedKey::ArrowRight) => self.view_ctx.seek_forward(10),
+                            Key::Named(NamedKey::ArrowLeft) =>  self.view_ctx.seek_backwards(10),
+                            _ => {}
+                        }
+                    } else {
+                        match event.logical_key {
+                            Key::Named(NamedKey::ArrowRight) => self.view_ctx.seek_forward(1),
+                            Key::Named(NamedKey::ArrowLeft) =>  self.view_ctx.seek_backwards(1),
+                            _ => {}
+                        }
+                    }
```

Comment:
On my local copy I added a `render_state.window.request_redraw()` here so that the page change shows up.